packages/perseus/src/widgets/__tests__/radio_test.jsx

```diff
@@ -632,3 +842,94 @@ describe("multi-choice question", () => {
         },
     );
 });
+
+describe("randomized question", () => {
+    const apiOptions: APIOptions = {
+        crossOutEnabled: false,
+    };
+    const [question] = multiChoiceQuestionAndAnswer;
+    const radio1Widget = ((question.widgets["radio 1"]: any): RadioWidget);
+    const radioOptions = radio1Widget.options;
+    const randomizeQuestion: PerseusRenderer = {
+        ...question,
+        widgets: {
+            ...question.widgets,
+            "radio 1": ({
+                ...radio1Widget,
+                options: {
+                    ...radioOptions,
+                    choices: [
+                        {content: "$x=-6$", correct: false},
+                        {content: "$x=4$", correct: true},
+                        {content: "$x=7$", correct: false},
+                        {
+                            content: "There is no such input value.",
+                            isNoneOfTheAbove: true,
+                            correct: false,
+                        },
+                    ],
+                    randomize: true,
+                },
+            }: RadioWidget),
+        },
+    };
+
+    beforeEach(() => {
+        jest.spyOn(Dependencies, "getDependencies").mockReturnValue(
+            testDependencies,
+        );
+    });
+
+    it("should snapshot the same after randomization based on the same seed", () => {
+        // Arrange / Act
+        const {container} = renderQuestion(randomizeQuestion, apiOptions);
+
+        // Assert
+        expect(container).toMatchSnapshot("first render");
+    });
+
+    it("should select and score correctly after randomization", () => {
+        // Arrange / Act
+        const {renderer} = renderQuestion(randomizeQuestion, apiOptions);
+        const option = screen.getByRole("checkbox", {name: /x=4/});
+        userEvent.click(option);
+
+        // Assert
+        expect(renderer).toHaveBeenAnsweredCorrectly();
+    });
+
+    it("should deselect as expected after randomization", () => {
+        // Arrange / Act
```

Comment:
**nit**: I don't think this line is really the 'Act' piece.  When building out a test I try to annotate a single line with `// Act` to denote exactly which action in the test is being exercised.  I suspect the `Act` is actually line 906 where we click to deselect.  Everything after that would be `// Assert` (including the `getAllByRole("checkbox")` on line 907.

Reply:
For sure. Thank you for pointing out these testing best practices! I updated it as you described, but I'm wondering, would it look cleaner to have options assigned during the Arrange section? The variable should still point to all the available options before and after the clicking option. Just wondering as it feels strange having an assignment statement in the Assert area; maybe that happens more than I think though? Looking at it again, maybe it does make more logical sense / create a better flow at this point in the test though.